src/btree/bt_debug.c

```diff
@@ -951,11 +971,15 @@ __wt_debug_tree(void *session_arg, WT_BTREE *btree, WT_REF *ref, const char *ofi
  *     Dump the in-memory information for a page.
  */
 int
-__wt_debug_page(void *session_arg, WT_BTREE *btree, WT_REF *ref, const char *ofile)
+__wt_debug_page(
+  void *session_arg, WT_BTREE *btree, WT_REF *ref, const char *ofile, bool dump_app_data)
 {
     WT_DBG *ds, _ds;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
+    uint32_t flags;
+
+    flags = dump_app_data ? WT_DEBUG_DUMP_APP_DATA | WT_DEBUG_TREE_LEAF : WT_DEBUG_TREE_LEAF;
```

Comment:
I'd be inclined to rewrite this as:

```
flags = WT_DEBUG_TREE_LEAF;
if (dump_app_data)
    flags |= WT_DEBUG_DUMP_APP_DATA;
```

Mostly because I don't like ternary statements very much, and the repeat of `WT_DEBUG_TREE_LEAF` doesn't feel clean.